File: php/bottle/Bottle.php
<?php

class Bottles
{
    public function song()
    {
        return $this->verses(99, 0);
    }

    public function verses(int $hi, int $lo)
    {
        $poem = '';
        for ( $n = $hi; $n >= $lo; $n--) {
            $poem .=$this->verse($n);
            $poem .= '</br>';
        }

        return $poem;
    }

    public function verse(int $n)
    {
        $verse = '';
        $verse .=  ($n == 0 ? 'No more' : $n) . ' bottle' . ($n == 1 ? '' : 's') . ' of beer on the wall. ';
        $verse .=  ($n == 0 ? 'No more' : $n) . ' bottle' . ($n == 1 ? '' : 's') . ' of beer.';
        $verse .= '</br>';
        $verse .=  $n > 0 ? ('Take ' . ($n > 1 ? 'one' : 'it') . ' down and pass it around, ') : 'Go to the store and buy some more, ';
        $verse .=  ($n - 1 < 0 ? 99 : ($n - 1 == 0 ? ' no more ' : $n - 1)) . ' bottle' . ($n == 1 ? '' : 's') . ' of beer on the wall.';
        $verse .= '</br>';

        return $verse;
    }
}

File: java/src/com/framgia/Part1/bottle/Bottle.java
package com.framgia.Part1.bottle;

public class Bottle {

    private static String song() {
        return getVerses(99, 0);
    }

    private static String getVerses(int hi, int lo) {
        StringBuilder poem = new StringBuilder();
        for (int n = hi; n >= lo; n--) {
            poem.append(getVerse(n));
            poem.append("\n");
        }
        return poem.toString();
    }

    private static String getVerse(int n) {
        String verse = "";
        verse += n == 0 ? "No more bottles" : (n == 1) ? "1 bottle" : n + " bottles";
        verse += " of beer on the wall, ";
        verse += n == 0 ? "No more bottles" : (n == 1) ? "1 bottle" : n + " bottles" + " of beer.";
        verse += "\n";
        verse += n == 1 ? "Go to the store and buy some more, " : "Take one down and pass it around, ";
        verse += n == 1 ? "No more bottles" : (n == 2) ? "1 bottle" : n == 0 ? "99 bottles" : (n - 1) + "bottles";
        verse += " of beer on the wall.";
        return verse;
    }

    public static void main(String[] args) {
        System.out.println(song());
    }
}
